test/fixtures/plugin-mongodb-core.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+
+var traceUtil = require('../../src/util.js');
+var shimmer = require('shimmer');
+var semver = require('semver');
+
+var api;
+
+var SUPPORTED_VERSIONS = '1 - 2';
+
+/**
+ * Wraps the provided callback so that the provided span will
+ * be closed immediately after the callback is invoked.
+ *
+ * @param {Span} span The span to be closed.
+ * @param {Function} done The callback to be wrapped.
+ * @return {Function} The wrapped function.
+ */
+function wrapCallback(transaction, done) {
+  return function(err, res) {
+    if (api.config.enhancedDatabaseReporting) {
+      if (err) {
+        // Errors may contain sensitive query parameters.
+        transaction.addLabel('mongoError', err);
+      }
+      if (res) {
+        var result = res.result ? res.result : res;
+        transaction.addLabel('results', traceUtil.stringifyPrefix(
```

Comment:
I agree.  This method seems like it will be used by a lot of plugins.  Another approach could be to have a small set of core functionality in `trace-plugin-interface.js` and have another helper class that accepts an `api` object and has methods that perform common complex tasks using the `api` object.

This could allow the helper to be tested independent of the api's core functionality by mocking the api.